components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/DefaultIdentityProviderImpl.java

```diff
@@ -87,6 +91,54 @@ public boolean isValidRole(String roleName) {
     }
 
     @Override
+    public List<String> getRoleIdsOfUser(String userId) throws IdentityProviderException {
+        List<String> roleIds = new ArrayList<>();
+        SCIMUser scimUser = scimServiceStub.getUser(userId);
+        if (scimUser != null) {
+            List<SCIMUser.SCIMUserGroups> roles = scimUser.getGroups();
+            if (roles != null) {
+                roles.forEach(role -> roleIds.add(role.getValue()));
+            }
+        } else {
+            String errorMessage = "User id " + userId + " does not exist in the system.";
+            log.error(errorMessage);
+            throw new IdentityProviderException(errorMessage, ExceptionCodes.USER_DOES_NOT_EXIST);
+        }
+        return roleIds;
+    }
+
+    @Override
+    public String getRoleId(String roleName) throws IdentityProviderException {
+        Response role = scimServiceStub.searchGroups(FILTER_PREFIX + roleName);
+        String roleId = null;
+        if (role.status() == 200) {
+            String responseBody = role.body().toString();
+            JsonParser parser = new JsonParser();
+            JsonObject parsedResponseBody = (JsonObject) parser.parse(responseBody);
+            JsonArray roleList = (JsonArray) parsedResponseBody.get("Resources");
+            if (roleList.size() == 1) {
```

Comment:
Usually in SCIM, it doesn't allow a role with the same display name be created. We assume that role name is unique. Hence usually a single role will be returned. But precaution is taken to check whether the list actually contains a single role. Else a null roleId will be returned. That's why we check for one.